File: regexlib/javascript_test_file/regexlib_2751.js
/* 2751
 * (([0-9]|[ ]|[-]|[\(]|[\)]|ext.|[,])+)([ ]|[:]|\t|[-])*(Home|Office|Work|Away|Fax|FAX|Phone)|(Home|Office|Work|Away|Fax|FAX|Phone|Daytime|Evening)([ ]|[:]|\t|[-])*(([0-9]|[ ]|[-]|[\(]|[\)]|ext.|[,])+)|(([(]([0-9]){3}[)]([ ])?([0-9]){3}([ ]|-)([0-9]){4}))
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"0"*5000+"!1 _SLQ_1"
 */
var REGEX = /(([0-9]|[ ]|[-]|[\(]|[\)]|ext.|[,])+)([ ]|[:]|\t|[-])*(Home|Office|Work|Away|Fax|FAX|Phone)|(Home|Office|Work|Away|Fax|FAX|Phone|Daytime|Evening)([ ]|[:]|\t|[-])*(([0-9]|[ ]|[-]|[\(]|[\)]|ext.|[,])+)|(([(]([0-9]){3}[)]([ ])?([0-9]){3}([ ]|-)([0-9]){4}))/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '0'.repeat(i*10000) + '!1 _SLQ_1'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_2751.py
# 2751
# (([0-9]|[ ]|[-]|[\(]|[\)]|ext.|[,])+)([ ]|[:]|\t|[-])*(Home|Office|Work|Away|Fax|FAX|Phone)|(Home|Office|Work|Away|Fax|FAX|Phone|Daytime|Evening)([ ]|[:]|\t|[-])*(([0-9]|[ ]|[-]|[\(]|[\)]|ext.|[,])+)|(([(]([0-9]){3}[)]([ ])?([0-9]){3}([ ]|-)([0-9]){4}))
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"0"*5000+"!1 _SLQ_1"

import re2 as re
from time import perf_counter

regex = """(([0-9]|[ ]|[-]|[\(]|[\)]|ext.|[,])+)([ ]|[:]|\t|[-])*(Home|Office|Work|Away|Fax|FAX|Phone)|(Home|Office|Work|Away|Fax|FAX|Phone|Daytime|Evening)([ ]|[:]|\t|[-])*(([0-9]|[ ]|[-]|[\(]|[\)]|ext.|[,])+)|(([(]([0-9]){3}[)]([ ])?([0-9]){3}([ ]|-)([0-9]){4}))"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "0" * i * 10000 + "!1 _SLQ_1"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")